pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -121,17 +121,28 @@ func (v *fleetResourceValidator) handleMemberCluster(req admission.Request) admi
 	return validation.ValidateUserForResource(req, v.whiteListedUsers)
 }
 
-// handleInternalMemberCluster allows/denies the request to modify internal member cluster object after validation.
-func (v *fleetResourceValidator) handleInternalMemberCluster(ctx context.Context, req admission.Request) admission.Response {
-	if req.Operation == admissionv1.Update && req.SubResource == "status" {
-		return validation.ValidateMCIdentity(ctx, v.client, req, req.Name)
+// handleFleetMemberNamespacedResource allows/denies the request to modify object after validation.
+func (v *fleetResourceValidator) handleFleetMemberNamespacedResource(ctx context.Context, req admission.Request) admission.Response {
+	var response admission.Response
+	if isFleetMemberNamespace(req.Namespace) {
+		// check to see if valid users other than member agent is making the request.
+		response = validation.ValidateUserForResource(req, v.whiteListedUsers)
+		// check to see if member agent is making the request only on Update.
+		if !response.Allowed && req.Operation == admissionv1.Update {
+			mcName := parseMemberClusterNameFromNamespace(req.Namespace)
+			if mcName != "" {
+				response = validation.ValidateMCIdentity(ctx, v.client, req, mcName)
+			}
```

Comment:
Hi Arvind! It might be better to record an error here; this should be an unexpected case I assume.